internal/framework/status/updater.go

```diff
@@ -169,3 +182,43 @@ func (upd *updaterImpl) update(
 			"kind", obj.GetObjectKind().GroupVersionKind().Kind)
 	}
 }
+
+func UpdateObjStatus(
+	ctx context.Context,
+	attempts int,
+	sleep time.Duration,
+	obj client.Object,
+	upd *UpdaterImpl,
+) (err error) {
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			time.Sleep(sleep)
+			sleep *= 2
+		}
+		if err = upd.cfg.Client.Status().Update(ctx, obj); err == nil {
```

Comment:
What if the context is canceled during this loop? We want to make sure that we return as soon as possible when the context is canceled.